import/popups/PopupOpener.qml

```diff
@@ -189,6 +154,127 @@ Timer {
         return true
     }
 
+    // Open alert dialog
+    function alert(data) {
+        var obj = pageStack.animatorPush(Qt.resolvedUrl("AlertDialog.qml"), {
+            "text": data.text,
+            "checkbox": getCheckbox(data)
+        })
+        obj.pageCompleted.connect(function(dialog) {
+            // TODO: also the Async message must be sent when window gets closed
+            dialog.done.connect(function() {
+                contentItem.sendAsyncMessage("alertresponse", {
+                    "winId": data.winId,
+                    "checkvalue": dialog.checkboxValue
+                })
+            })
+        })
+    }
+
+    // Open confirm dialog
+    function confirm(data) {
+        var obj = pageStack.animatorPush(Qt.resolvedUrl("ConfirmDialog.qml"), {
+            "text": data.text,
+            "checkbox": getCheckbox(data)
+        })
+        obj.pageCompleted.connect(function(dialog) {
+            // TODO: also the Async message must be sent when window gets closed
+            dialog.accepted.connect(function() {
+                contentItem.sendAsyncMessage("confirmresponse", {
+                    "winId": data.winId,
+                    "accepted": true,
+                    "checkvalue": dialog.checkboxValue
+                })
+            })
+            dialog.rejected.connect(function() {
+                contentItem.sendAsyncMessage("confirmresponse", {
+                    "winId": data.winId,
+                    "accepted": false,
+                    "checkvalue": dialog.checkboxValue
+                })
+            })
+        })
+    }
+
+    // Open prompt dialog
+    function prompt(data) {
+        var obj = pageStack.animatorPush(Qt.resolvedUrl("PromptDialog.qml"), {
+            "text": data.text,
+            "value": data.defaultValue,
+            "checkbox": getCheckbox(data)
+        })
+        obj.pageCompleted.connect(function(dialog) {
+            // TODO: also the Async message must be sent when window gets closed
+            dialog.accepted.connect(function() {
+                contentItem.sendAsyncMessage("promptresponse", {
+                    "winId": data.winId,
+                    "accepted": true,
+                    "promptvalue": dialog.value,
+                    "checkvalue": dialog.checkboxValue
+                })
+            })
+            dialog.rejected.connect(function() {
+                contentItem.sendAsyncMessage("promptresponse", {
+                    "winId": data.winId,
+                    "accepted": false,
+                    "checkvalue": dialog.checkboxValue
+                })
+            })
+        })
+    }
+
+    // Open login dialog
+    function login(data) {
+        var obj = pageStack.animatorPush(Qt.resolvedUrl("PasswordManagerDialog.qml"), {
```

Comment:
```suggestion
        pageStack.animatorPush(Qt.resolvedUrl("PasswordManagerDialog.qml"), {
```

Reply:
Please add also testcase for https://github.com/sailfishos/sailfish-browser/blob/next/tests/manual/testwebprompts.html. Simple testcase function that requests few alerts, prompts, and confirms.